topcoffea/modules/selection.py

```diff
@@ -335,3 +335,12 @@ def addLepCatMasks(events):
     events['is_emmm'] = ((n_e_4l==1) & (n_m_4l==3))
     events['is_mmmm'] = ((n_e_4l==0) & (n_m_4l==4))
     events['is_gr4l'] = ((n_e_4l+n_m_4l)>4)
+
+
+# Returns a mask for events with a same flavor opposite sign pair close to the Z
+def get_Z_peak_mask(lep_collection):
+    ll_pairs = ak.combinations(lep_collection, 2, fields=["l0","l1"])
+    zpeak_mask = (abs((ll_pairs.l0+ll_pairs.l1).mass - 91.2)<10.0) 
```

Comment:
I believe this is only checking 0 and lepton 1 (unless I'm not thinking of the awkward arrays correctly). Don't we also need to check l0+l2 and l1+l2 for the 3l case?

Reply:
Yes, we definitely want to form all of the possible pairs, but I think that's what we're doing (at least that's what I'm trying to do anyway). I think the `l0` and `l1` should just be labels for accessing the first and second element in each pair. So for example:
```
import awkward as ak
my_arr = ak.Array([["a","b","c"],["d","e"],["f"],[]])
my_pairs = ak.combinations(my_arr, 2, fields=["l0","l1"])
for pair in my_pairs:
    print(pair)
```
Gives this:
```
[{l0: 'a', l1: 'b'}, {l0: 'a', l1: 'c'}, {l0: 'b', l1: 'c'}]
[{l0: 'd', l1: 'e'}]
[]
[]
```
But I'm not sure if this addresses your concern, or if I'm missing something about the question?

Reply:
Thanks for that example. So I was just thinking about awkward arrays incorrectly, and all pairs are considered.